conseil-common/src/test/scala/tech/cryptonomic/conseil/common/rpc/RpcClientTest.scala

```diff
@@ -0,0 +1,106 @@
+package tech.cryptonomic.conseil.common.rpc
+
+import scala.concurrent.ExecutionContext
+
+import cats.effect._
+import org.http4s._
+import org.http4s.client.Client
+import fs2.Stream
+import io.circe.generic.auto._
+import org.http4s.circe.CirceEntityDecoder._
+import org.http4s.circe.CirceEntityEncoder._
+import org.scalatest.{Matchers, WordSpec}
+
+import tech.cryptonomic.conseil.common.rpc.RpcClient._
+
+class RpcClientTest extends WordSpec with Matchers {
+
+  implicit val contextShift: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
+
+  "Rpc Client" should {
```

Comment:
We are testing `json` here, because that's what we are supporting now, or we support whatever we can, and we simply choose json here? Maybe, we can add some tests for plain-text also. If that does not make sense, then, response should always return `Json` not `R`.

Reply:
This is an implementation of the JSON-RPC specification :) Json is just the underlying protocol in my opinion